File: DDSApp/Models/Doc.cs
using System.Collections.Generic;
using DDSApp.Areas.Abstractions;


namespace DDSApp.Models
{
    public class Doc : IEntityBase
    {
        public string Id { get; set; } 
        public string Title { get; set; }
        public string Topic { get; set; }
        public string Category { get; set; }
        public string Url { get; set; }
        public string LastUpdate { get; set; }
    }
}

File: DDSApp/ClientApp/src/reducers/doc.js
import { createReducer } from 'redux-act'

import * as a from '../actions/doc'

const getDefaultState = () => ({
    documentId: undefined,
    title: undefined,
    topic: undefined,
    category: undefined,
    url: undefined,
    lastUpdate: undefined,
})

export default _ =>
    createReducer(
        {
            [a.receiveDoc]: (state, doc) => ({
                ...state,
                ...doc,
                documentId: doc.id,
                title: doc.title,
                topic: doc.topic,
                category: doc.category,
                url: doc.url,
                lastUpdate: doc.lastUpdate,
            }),
        },
        getDefaultState()
    )
